webdriver/tests/document_handling/page_source.py

```diff
@@ -0,0 +1,74 @@
+import pytest
+import json
+
+from tests.support.asserts import assert_error, assert_success, assert_dialog_handled
+from tests.support.fixtures import create_dialog
+from tests.support.inline import inline
+
+
+def test_closed_browsing_context(new_session):
+    _, session = new_session({})
+    session.end()
```

Comment:
You want `create_window` and then get the page source from the closed browsing context. That will return `no such window`. This is testing that you can’t get the page source of an ended session, which would return `session not created`.